dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java

```diff
@@ -536,6 +539,33 @@ private static List<String> getDistinctEngineVersions(final Set<Version> workflo
             .toList();
     }
 
+    /**
+     * Returns a set of Author containing non-ORCID authors and ORCID authors with additional information.
+     * @param entry
+     * @return
+     */
+    private static Set<Author> getAllAuthors(Entry entry) {
+        Set<Author> allAuthors = new HashSet<>();
+        if (!entry.getOrcidAuthors().isEmpty()) {
+            Optional<String> token = ORCIDHelper.getOrcidAccessToken();
+            if (token.isPresent()) {
+                Set<OrcidAuthorInformation> orcidAuthorInformation = ((Set<OrcidAuthor>)entry.getOrcidAuthors()).stream()
+                        .map(orcidAuthor -> ORCIDHelper.getOrcidAuthorInformation(orcidAuthor.getOrcid(), token.get()))
+                        .filter(Optional::isPresent)
+                        .map(Optional::get)
+                        .collect(Collectors.toSet());
+                allAuthors.addAll(orcidAuthorInformation);
+            }
+        }
+        allAuthors.addAll(entry.getAuthors());
+
+        if (allAuthors.isEmpty()) {
+            // Add an empty author with a null name so that ES has something to replace with the null_value otherwise an empty array is ignored by ES
```

Comment:
If there are no authors, wouldn't we want it ignored by ES? Or does that cause problems with the UI if it's in some results but not others?

Reply:
We don't want to ignore it because we want to have a count for `n/a` authors in the UI (related ticket https://github.com/dockstore/dockstore/issues/4324)